### ISSUE TITLE
Better/old concrete style

### ISSUE THREAD

**Issue by ST-DDT** [jpanther/Dectorio#51]
Could you add alternative images for concrete to your mod? (Big tiles instead of small slates)
Similar to the old ones (But hopefully in hr). 

Because the current concrete looks like it was placed by a lazy gardener quite a bit in the past. Or like being ploughed up by a tank.

The borders that were added in 0.16 look (almost) nice even when mixed with the old 0.15 concrete images, however there is too few contrast between both. I guess it would be sufficient to change the color of the main concrete image.

I tried it myself using this code:

````lua
data.raw.tile["concrete"].material_background.picture = "__MyConcrete__/graphics/terrain/concrete/concrete.png"
data.raw.tile["concrete"].hr_version.material_background.picture = "__MyConcrete__/graphics/terrain/concrete/hr-concrete.png"

data.raw.tile["hazard-concrete-left"].material_background.picture = "__MyConcrete__/graphics/terrain/hazard-concrete-left/concrete.png"
data.raw.tile["hazard-concrete-left"].hr_version.material_background.picture = "__MyConcrete__/graphics/hazard-concrete-left/concrete/hr-concrete.png"

data.raw.tile["hazard-concrete-right"].material_background.picture = "__MyConcrete__/graphics/terrain/hazard-concrete-right/concrete.png"
data.raw.tile["hazard-concrete-right"].hr_version.material_background.picture = "__MyConcrete__/graphics/terrain/hazard-concrete-right/hr-concrete.png"
````

And some slightly edited files from 0.15.
![concrete](https://user-images.githubusercontent.com/1579362/34122147-bc51df5a-e42b-11e7-9c95-712011edd7bd.png)

However they look blurry due to the zooming.
My own attemps to create better looking/HR concrete went horribly wrong.

**Comment by jpanther** [comment]
I've just been digging around the 0.16 assets and have found one called `tutorial-grid` which looks rather concrete-like. This is in hr and could be a really good base for a grid style concrete. I'll mock up a quick item that allows it to be placed in-game and see what it looks like. The only downside I can see right now is there's no edge transitions so it will end rather abruptly, but that can be fixed!

![hr-tutorial-grid1](https://user-images.githubusercontent.com/4462786/34130536-4eef8f36-e49c-11e7-8e65-883cc9809458.png)

**Comment by ST-DDT** [comment]
The classic look is not really important for me. 
I just want 

* [x] "clean" (no grass decorations/leftover rocks), 
* [ ] "tidy" (non-hazardous to walk on) 
* [ ] and "big" concreate tiles (1x1 (or 2x2) per m², like in some real world factories ).

**Comment by ST-DDT** [comment]
> The only downside I can see right now is there's no edge transitions so it will end rather abruptly, but that can be fixed!

You don't need the edge transitions, just use the corner stones from the new concrete.
(simply swap the graphics of the old and everything might work as is.)

**Comment by jpanther** [comment]
Yeah I'll see how that goes first. I have a feeling these assets aren't large enough to be fed into that directly but there's very little documentation on the new tilesets at the moment so it's all a guessing game.

**Comment by ST-DDT** [comment]
You could try my code snippet from above for the start. Maybe that is all that is needed. (maybe you have to copy the image a bit to match the required size.) 

Good luck.

**Comment by jpanther** [comment]
Yeah, as I suspected it doesn't like this size asset for their new 'material' sprites.

![screen shot 2017-12-19 at 9 18 06](https://user-images.githubusercontent.com/4462786/34130866-8e4005de-e49d-11e7-9c8f-1aa7e2782468.png)

A bit more tweaking and I got the game to start but it's doing something very weird and placing random parts of the spritesheet when I put down the tile! Very funky!

![screen shot 2017-12-19 at 9 22 10](https://user-images.githubusercontent.com/4462786/34131026-1e382a5e-e49e-11e7-9de9-8312e81c6a29.png)

You can see a strip of the concrete part poking through and it doesn't look bad. I think it looks good enough for me to pursue getting this to work properly.

**Comment by jpanther** [comment]
I've got this working now, with only a really basic edge mask specified...

![screen shot 2017-12-19 at 9 58 45](https://user-images.githubusercontent.com/4462786/34132294-4d6dbc6c-e4a3-11e7-8dac-e62f19935fe7.png)

With a rougher edge, it looks like this...

![screen shot 2017-12-19 at 10 01 06](https://user-images.githubusercontent.com/4462786/34132365-91eef1a8-e4a3-11e7-9dd1-4e62e4fe8463.png)

And then with the 'brick' edging from the new 0.16 concrete...

![screen shot 2017-12-19 at 10 04 46](https://user-images.githubusercontent.com/4462786/34132470-1c96bb92-e4a4-11e7-9df7-d8299efc1487.png)

What do you think about any of these? I actually think that last one looks okay.

**Comment by jpanther** [comment]
At the risk of this issue becoming somewhat of a dev blog... I've done a bit more tweaking to the colours and mocked it up with some other terrains to see how it fits into the Factorio world. I think this isn't bad. Even the hazard one layers on top reasonably well (although you do still have to put up with the bricks sticking through).

![screen shot 2017-12-19 at 10 58 51](https://user-images.githubusercontent.com/4462786/34134047-6b24d0ac-e4ac-11e7-993b-0d64e59ac60f.png)

**Comment by jpanther** [comment]
I've made a bunch more changes including more variation on the grid concrete, darkened it some more, and toned down all the normal painted concrete styles so they don't look so 'bricky'. This way they can be layered without them looking too ridiculous. I've pushed all these into Dectorio 0.8.2, but let me know what you think!

![screen shot 2017-12-19 at 15 49 38](https://user-images.githubusercontent.com/4462786/34141393-02365ff6-e4d5-11e7-9b06-07d2ea3f06ee.png)

Ref: e1874e4c9d7fa474b0fea6d273d12eec7e2c92c6
